PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobFactory.java

```diff
@@ -0,0 +1,213 @@
+                    logger.log().warn("Error getting access token based on the given ADLS root path. " + ExceptionUtils.getStackTrace(ex));
+                    throw new ExecutionException("Error getting access token based on the given ADLS root path");
+                }
+
+                break;
+            case ADLS_GEN2:
+                destinationRootPath = submitModel.getJobUploadStorageModel().getUploadPath();
+                String gen2StorageAccount = "";
+                Matcher m = Pattern.compile(SparkBatchJob.AdlsGen2RestfulPathPattern).matcher(destinationRootPath);
+                if(m.find()){
+                    gen2StorageAccount = m.group("accountName");
+                }
+
+                if (StringUtils.isBlank(gen2StorageAccount)) {
+                    throw new ExecutionException("Invalid ADLS GEN2 root path.");
+                }
+
+                accessKey = submitModel.getJobUploadStorageModel().getAccessKey();
+                if (StringUtils.isBlank(accessKey)) {
+                    throw new ExecutionException("Invalid access key input");
+                }
+
+                httpObservable = new SharedKeyHttpObservable(gen2StorageAccount, accessKey);
+                jobDeploy = new ADLSGen2Deploy(httpObservable);
+                break;
+            case WEBHDFS:
+                destinationRootPath = submitModel.getJobUploadStorageModel().getUploadPath();
+                if (StringUtils.isBlank(destinationRootPath) || !destinationRootPath.matches(SparkBatchJob.WebHDFSPathPattern)) {
+                    throw new ExecutionException("Invalid webhdfs root path input");
+                }
+
+                //create httpobservable and jobDeploy
+                try {
+                    if (clusterDetail instanceof ClusterDetail) {
+                        httpObservable = new AzureHttpObservable(clusterDetail.getSubscription().getTenantId(), ApiVersion.VERSION);
+                        jobDeploy = clusterDetail.getStorageAccount().getAccountType() == StorageAccountTypeEnum.ADLS
+                                ? new ADLSGen1HDFSDeploy(clusterDetail, httpObservable)
+                                : null;
+                    } else if (clusterDetail instanceof SqlBigDataLivyLinkClusterDetail) {
+                        httpObservable = new HttpObservable(clusterDetail.getHttpUserName(), clusterDetail.getHttpPassword());
+                        jobDeploy = new WebHDFSDeploy(clusterDetail, httpObservable);
+                    }
+                } catch (HDIException ignore) {
+                }
+
+                if (httpObservable == null || jobDeploy == null) {
+                    throw new ExecutionException("Error preparing webhdfs uploading info based on the given cluster");
+                }
+
+                break;
+        }
+
+        //TODO:use httpobservable to replace sparkbathsubmission and deprecate the old constructor.
+        if (httpObservable != null) {
```

Comment:
I suppose it might be better to the following new instance operations into the separated `buildSparkBatchJob` methods